### PR TITLE
Feature/hide occluded studs

### LINKED ISSUE #652: Hide studs if they are covered by the next brick layer
A BrickObject's studs can be set to invisible if we know that they are hidden under the next brick. This should improve performance.

### PR DESCRIPTION
**Pull request by AtiX**
Closes #652

### CODE REVIEW

File: src/plugins/nodeVisualizer/visualization/brickVisualization.coffee [hunk 1]
```diff
@@ -265,6 +277,11 @@ class BrickVisualization
 
 		for voxel in voxels
 			voxel.make3dPrinted()
+			# show studs of brick below
```
Comment (adius) [inline src/plugins/nodeVisualizer/visualization/brickVisualization.coffee]: `# Show`

**Comment by AtiX** [comment]
Vanderbilt: 11fps old versus 22fps new :trumpet:

File: src/plugins/nodeVisualizer/visualization/brickVisualization.coffee [hunk 2]
```diff
@@ -143,13 +149,17 @@ class BrickVisualization
...
-		# show not filled lego shape as outline
+		# Show not filled lego shape as outline
 		outlineCoords = @printVoxels.map (voxel) -> voxel.position
 		@voxelWireframe.createWireframe outlineCoords
 
 		@_visibleChildLayers = null
 
-		#ToDo: hide studs when brick is completely below other brick
+	_setStudVisibility: (brick) ->
+		if brick.isCoveredOnTop()
```
Comment (stuikomma) [inline src/plugins/nodeVisualizer/visualization/brickVisualization.coffee]: Can also be done without if/else.

**Comment by stuikomma** [comment]
:airplane:

**Pull request merged**